src/models/connectionStore.ts

```diff
@@ -294,8 +294,14 @@ export class ConnectionStore {
      * Clear all recently used connections from the MRU list.
      */
     public async clearRecentlyUsed(): Promise<void> {
-        // Update the MRU list to be empty
         try {
+            // Get all recent connection profiles and delete the associated credentials.
+            const mruList = this.getRecentlyUsedConnections();
+            for (const connection of mruList) {
+                const credentialId = ConnectionStore.formatCredentialId(connection.server, connection.database, connection.user, ConnectionStore.CRED_MRU_USER);
+                await this._credentialStore.deleteCredential(credentialId);
```

Comment:
We should have each connection be wrapped in a try/catch so that if one fails to be removed we still delete the others. And for ones that error out we have a choice : 

1. Leave it in the MRU (since we couldn't delete the credential, this gives the person another chance to remove it from ADS in case it was some weird transient error)
2. Still clear the MRU at the end but maybe warn them we couldn't delete it so the user can go clean up the credential themself? Not sure how we expect them to find it easily though.

I'm slightly in favor of 2 since that seems like a better user experience if the user actually wants to clear out their MRU - and this is pretty unlikely to happen anyways.